product_pricelist_overcharge_by_amount/models/product_pricelist.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2024 Binhex - Christian Ramos
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare, float_round
+
+
+class ProductPricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    def _compute_price_rule(self, products_qty_partner, date=False, uom_id=False):
+        """Apply overcharge to the price when applicable."""
+        rule_obj = self.env["product.pricelist.item"]
+        result = super()._compute_price_rule(products_qty_partner, date, uom_id)
+        for product, _qty, _partner in products_qty_partner:
+            rule = rule_obj.browse(result[product.id][1])
+            if rule.overcharge:
+                for overcharge in rule.overcharge_item_ids:
+                    price = result[product.id][0]
+                    if overcharge.can_be_apply(price):
+                        result[product.id] = (
+                            overcharge.apply_overcharge(price),
+                            rule.id,
+                        )
+                        break
+        return result
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    overcharge = fields.Boolean(
+        string="Apply overcharge to the final price",
+    )
+    overcharge_item_ids = fields.One2many(
+        "product.pricelist.overcharge.item",
+        "item_id",
+        "Overcharge Items",
+        help="Only match prices from the selected supplier",
+        copy=True,
+    )
+
+
+class ProductPricelistOverchargeItem(models.Model):
+    _name = "product.pricelist.overcharge.item"
```

Comment:
@Christian-RB Please split one model per file. Thanks

Reply:
@rousseldenis Thanks for the review and apologies for the delayed response. AFAIK if the model is used as a `One2many` field in the main model, it was acceptable to include it in the same file. Has this approach changed?